pkg/scheduler/apis/config/v1alpha1/conversion.go

```diff
@@ -20,11 +20,30 @@ import (
 	"fmt"
 
 	"k8s.io/apimachinery/pkg/conversion"
+	"k8s.io/apimachinery/pkg/util/validation/field"
 	"k8s.io/kube-scheduler/config/v1alpha1"
 	"k8s.io/kubernetes/pkg/scheduler/apis/config"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins"
+	"k8s.io/kubernetes/pkg/scheduler/framework/plugins/interpodaffinity"
 )
 
-// Convert_v1alpha1_KubeSchedulerLeaderElectionConfiguration_To_config_KubeSchedulerLeaderElectionConfiguration is an autogenerated conversion function.
+// Convert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfiguration converts to the internal.
+func Convert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfiguration(in *v1alpha1.KubeSchedulerConfiguration, out *config.KubeSchedulerConfiguration, s conversion.Scope) error {
+	if err := autoConvert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfiguration(in, out, s); err != nil {
+		return err
+	}
+	if in.HardPodAffinitySymmetricWeight != nil {
+		if err := interpodaffinity.ValidateHardPodAffinityWeight(field.NewPath("hardPodAffinitySymmetricWeight"), *in.HardPodAffinitySymmetricWeight); err != nil {
```

Comment:
doing validation in conversion is not ideal. can we make this error-free and validate the weight when validating plugin config?

Reply:
Removed. Note that we do this validation during framework instantiation.
I opened #88093 to track this.